redux/cartSlice.ts

```diff
@@ -7,29 +7,29 @@ const cartSlice = createSlice({
   },
   reducers: {
     addToCart: (state, action) => {
-      const itemInCart = state.cart.find(
+      const itemInCartIndex = state.cart.findIndex(
```

Comment:
Nice! Just found out that findIndex is indeed faster than find.

Reply:
`findIndex` is the right choice here, as I think `find` gives you back a copy of the object rather than a pointer to the object in the array.  I wish JS had iterators...